src/person/views.py

```diff
@@ -93,3 +93,38 @@ class MassMessageView(generics.GenericAPIView):
     def post(self, request):
         """Send custom push notification to multiple users by id."""
         return MassMessageController(request.data).process()
+
+
+class BlockUserView(generics.GenericAPIView):
+    serializer_class = UserSerializer
+    permission_classes = api_settings.CONSUMER_PERMISSIONS
+
+    def post(self, request, id):
+        """Block user by id."""
+        if not User.objects.filter(id=id).exists():
+            return failure_response("User not found.", status.HTTP_404_NOT_FOUND)
+        if (
+            request.user.person.blocked_users.filter(id=id).exists()
+            or request.user.id == id
```

Comment:
should an error alert/message be generated if a user tries to block themselves instead of doing nothing

Reply:
User blocking themself should be an HTTP_403_FORBIDDEN, it doesn't make sense to block yourself and the frontend would not provide a way to do it so this route would never be invoked in that way unless someone is messing with our backend or frontend did something wrong - fix this